warehouse/macaroons/services.py

```diff
@@ -127,20 +127,33 @@ def verify(self, raw_macaroon, context, principals, permission):
 
         raise InvalidMacaroonError("invalid macaroon")
 
-    def create_macaroon(self, location, user_id, description, caveats):
+    def create_macaroon(
+        self, *, location, description, caveats, user_id=None, project_id=None
+    ):
         """
         Returns a tuple of a new raw (serialized) macaroon and its DB model.
         The description provided is not embedded into the macaroon, only stored
         in the DB model.
+
+        Either a user ID or a project ID (but not both) is required.
         """
-        user = self.db.query(User).filter(User.id == user_id).one()
+        user = None
+        if user_id is not None:
+            user = self.db.query(User).get(user_id)
+
+        project = None
+        if project_id is not None:
+            project = self.db.query(Project).get(project_id)
```

Comment:
Is it actually necessary to query for the actual instance here? Or could we just initialize a `Macaroon` with the ID?